src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,180 @@
+
+    @Test
+    void register_userWasPutInList_Ok() {
+        registrationService.register(validUser);
+        StorageDao storageDao = new StorageDaoImpl();
+        String actual = storageDao.get(validUser.getLogin()).getLogin();
+        assertEquals(validUser.getLogin(), actual);
+    }
+
+    @Test
+    void register_returnCorrectUser_Ok() {
+        String actual = registrationService.register(validUser).getLogin();
+        assertEquals(validUser.getLogin(), actual);
+    }
+
+    @Test
+    void register_userAlreadyExists_NotOk() {
+        Storage.people.add(validUser);
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @Test
+    void register_userLoginLength_NotOk() {
+        validUser.setLogin("");
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(validUser);
+        });
+        validUser.setLogin("Use");
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(validUser);
+        });
+        validUser.setLogin("User1");
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @Test
+    void register_userLoginExactLength_Ok() {
+        validUser.setLogin("User12");
+        User actual = registrationService.register(validUser);
+        assertNotNull(actual);
+    }
+
+    @Test
+    void register_userLoginOverLength_Ok() {
+        validUser.setLogin("User12465786");
+        User actual = registrationService.register(validUser);
+        assertNotNull(actual);
+    }
+
+    @Test
+    void register_userPasswordExactLength_Ok() {
+        validUser.setPassword("123456");
+        User actual = registrationService.register(validUser);
+        assertNotNull(actual);
+    }
+
+    @Test
+    void register_userPasswordOverLength_Ok() {
+        validUser.setPassword("1234567890");
+        User actual2 = registrationService.register(validUser);
+        assertNotNull(actual2);
+    }
+
+    @Test
+    void register_userPasswordLength_NotOk() {
+        validUser.setPassword("");
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(validUser);
+        });
+        validUser.setPassword("123");
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(validUser);
+        });
+        validUser.setPassword("12345");
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @Test
+    void register_lowerThanAgeLimit_NotOk() {
+        validUser.setAge(15);
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(validUser);
+        });
+    }
+
+    @Test
+    void register_AgeLimit_Ok() {
+        validUser.setAge(18);
```

Comment:
![image](https://github.com/mate-academy/jv-registration-validation/assets/20326858/08657016-7fd1-48a7-ac1e-81b6cadee124)
**17 must be tested**